pkg/draft/context.go

```diff
@@ -65,3 +71,17 @@ func newAppContext(s *Server, req *rpc.UpRequest, out io.Writer) (*AppContext, e
 		vals: vals,
 	}, nil
 }
+
+func getulid() string { return <-ulidc }
+
+// A channel which returns build ulids.
+var ulidc = make(chan string)
+
+func init() {
+	rnd := rand.New(rand.NewSource(time.Now().UTC().UnixNano()))
+	go func() {
+		for {
+			ulidc <- ulid.MustNew(ulid.Timestamp(time.Now().UTC()), rnd).String()
```

Comment:
Isn't this a little excessive on clock cycles to be constantly generating ULIDs even when nothing is requesting one?

What if we just changed the function to

```
func getulid() string {
    rnd := rand.New(rand.NewSource(time.Now().UTC().UnixNano()))
    return ulid.MustNew(ulid.Timestamp(time.Now().UTC()), rnd).String()
}
```

Or is there some technical reason this isn't possible?